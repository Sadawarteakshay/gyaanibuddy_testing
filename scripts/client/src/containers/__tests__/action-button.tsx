import React from 'react';

import { render, renderApollo, cleanup, waitFor } from '../../test-utils';
import ActionButton from '../action-button';
import { cartItemsVar } from '../../cache';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('action button', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('render snapshot', () => {
    const ActionButtonElement = <ActionButton />;

    // see https://stackoverflow.com/questions/70805929/how-to-fix-error-usehref-may-be-used-only-in-the-context-of-a-router-compon
    render(ActionButtonElement, {wrapper: MemoryRouter});
    
    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ElementTree = renderer
      .create(<MemoryRouter>{ActionButtonElement}</MemoryRouter>)
      .toJSON();

    expect(ElementTree).toMatchSnapshot();
  });

  it('renders without error', () => {
    const { getByTestId } = renderApollo(<ActionButton />);
    expect(getByTestId('action-button')).toBeTruthy();
  });

  it('shows correct label', () => {
    const { getByText, container } = renderApollo(<ActionButton />);
    getByText(/add to cart/i);

    // rerender with different props to same container
    waitFor(() => cartItemsVar(['1']));
    renderApollo(<ActionButton id="1" />, { container });
    getByText(/remove from cart/i);
    cartItemsVar([]);

    // rerender with different props to same container
    renderApollo(<ActionButton isBooked={true} />, { container });
    getByText(/cancel this trip/i);
  });
});
