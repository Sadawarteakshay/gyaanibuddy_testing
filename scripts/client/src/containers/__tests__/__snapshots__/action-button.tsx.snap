// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action button render snapshot 1`] = `
<div>
  <button
    className="css-wwcn44"
    data-testid="action-button"
    onClick={[Function]}
  >
    Add to Cart
  </button>
</div>
`;
